---
title: "RNA Analysis 23"
author: "Michael Callahan"
date: "2024-04-03"
output: pdf_document
---

```{r setup, include=FALSE}
library(knitr)
knitr::opts_chunk$set(echo = TRUE)
```

## Load data

```{r Data load}
library(here)
library(dplyr)

current_project_path <- here()

file_path <- file.path(current_project_path, "Data", "CMP_RNA_1.tsv")
CMP_RNA_1 <- read.delim(file_path)

file_path <- file.path(current_project_path, "Data", "CMP_RNA_2.tsv")
CMP_RNA_2 <- read.delim(file_path)

file_path <- file.path(current_project_path, "Data", "CFU-E_RNA_1.tsv")
CFUE_RNA_1 <- read.delim(file_path)

file_path <- file.path(current_project_path, "Data", "CFU-E_RNA_2.tsv")
CFUE_RNA_2 <- read.delim(file_path)
```

## Data preprocessing

```{r preprocess}
#FYI - gene_ids are unique

#CREATE A TPM Dataframe
# Add prefixes to the tpm columns in each dataframe
CFUE_RNA_1_tpm <- CFUE_RNA_1 %>% select(gene_id, CFUE_RNA_1_tpm = TPM)
CFUE_RNA_2_tpm <- CFUE_RNA_2 %>% select(gene_id, CFUE_RNA_2_tpm = TPM)
CMP_RNA_1_tpm <- CMP_RNA_1 %>% select(gene_id, CMP_RNA_1_tpm = TPM)
CMP_RNA_2_tpm <- CMP_RNA_2 %>% select(gene_id, CMP_RNA_2_tpm = TPM)

# Join the dataframes together on gene_id
tpm <- inner_join(CFUE_RNA_1_tpm, CFUE_RNA_2_tpm, by = "gene_id") %>%
       inner_join(CMP_RNA_1_tpm, by = "gene_id") %>%
       inner_join(CMP_RNA_2_tpm, by = "gene_id") 

rm(CFUE_RNA_1_tpm, CFUE_RNA_2_tpm, CMP_RNA_1_tpm, CMP_RNA_2_tpm)

tpm <- tpm %>%
  rename_at(vars(2:5), ~ sub("_tpm$", "", .))


# Set 'column_name' as row names
rownames(tpm) <- tpm$gene_id

# Remove 'column_name' from dataframe (optional)
tpm <- tpm[, -which(names(tpm) == 'gene_id')]

#Create a colData matrix
colData = data.frame(
  source_name = c('CMP_RNA_1','CMP_RNA_2','CFUE_RNA_1','CFUE_RNA_2'),
  group = c('CMP', 'CMP', 'CFUE', 'CFUE')
  )
```

## Histogram overlay
Here, we can see a density plot of the 4 samples being compared.
```{r EDA}
library(RColorBrewer)
tpm_filtered <- tpm[rowSums(tpm != 0) > 0, ]
samplenames <- colnames(tpm_filtered)
tpm.cutoff <- log2(0.1)
nsamples <- ncol(tpm_filtered)
col <- brewer.pal(nsamples, "Paired")
par(mfrow=c(1,1))
plot(density(log(tpm_filtered[,1])), col=col[1], lwd=2, ylim=c(0,0.2), las=2, main="", xlab="")
title(main="TPM density", xlab="Log-TPM")
for (i in 2:nsamples){
den <- density(log(tpm_filtered[,i]))
lines(den$x, den$y, col=col[i], lwd=2)
}
legend("topright", samplenames, text.col=col, bty="n")
```


## EDA - Heat map
Here, we look at the TPM (trascripts per million) for our two cell lines (2 replicates each), and we pull out the top 100 genes with the highest variance of expression across samples. Then we plot their normalized expression levels across the samples using a heat map. We should see that the replicates within each cell line should have a more similar expression pattern than across cell lines.
```{r preprocess}
library(pheatmap)

#compute the variance of each gene across samples
variances <- apply(tpm, 1, var)

selectedGenes <- order(variances, decreasing = TRUE)[1:100]
pheatmap(tpm[selectedGenes,], scale = 'row', show_rownames = FALSE)
```

## EDA - PCA
...
```{r preprocess}
library(stats)
library(ggplot2)
library(ellipse)

M <- t(tpm[selectedGenes,])
M <- log2(M + 1)
pcaResults <- prcomp(M)

# Extract principal components from the PCA results
pc <- data.frame(PC1 = pcaResults$x[,1], PC2 = pcaResults$x[,2])

# Add sample metadata from colData
pc <- cbind(pc, colData)

# Plot PCA results using ggplot2
ggplot(pc, aes(x = PC1, y = PC2, color = group)) +
  geom_point() +
  labs(title = "PCA Plot", x = "PC1", y = "PC2")

summary(pcaResults)
```


## Correlation plots
```{r cor plots}
library(corrplot)
correlationMatrix <- cor(tpm)
corrplot(correlationMatrix, order = 'hclust', 
         addrect = 2, addCoef.col = 'white', 
         number.cex = 0.7) 
```


# Differential Expression Analysis
## Preprocessing
```{r diff ex preprocess}
#Get counts data
# Add prefixes to the tpm columns in each dataframe
CFUE_RNA_1_counts <- CFUE_RNA_1 %>% select(gene_id, CFUE_RNA_1_counts = expected_count)
CFUE_RNA_2_counts <- CFUE_RNA_2 %>% select(gene_id, CFUE_RNA_2_counts = expected_count)
CMP_RNA_1_counts <- CMP_RNA_1 %>% select(gene_id, CMP_RNA_1_counts = expected_count)
CMP_RNA_2_counts <- CMP_RNA_2 %>% select(gene_id, CMP_RNA_2_counts = expected_count)

# Join the dataframes together on gene_id
countData <- inner_join(CFUE_RNA_1_counts, CFUE_RNA_2_counts, by = "gene_id") %>%
       inner_join(CMP_RNA_1_counts, by = "gene_id") %>%
       inner_join(CMP_RNA_2_counts, by = "gene_id") 

rm(CFUE_RNA_1_counts, CFUE_RNA_2_counts, CMP_RNA_1_counts, CMP_RNA_2_counts)

countData <- countData %>%
  rename_at(vars(2:5), ~ sub("_counts$", "", .))

countData <- mutate_if(countData, is.numeric, round)

# Set 'column_name' as row names
rownames(countData) <- countData$gene_id

# Remove 'column_name' from dataframe (optional)
countData <- countData[, -which(names(countData) == 'gene_id')]

#Create a colData matrix
colData = data.frame(
  source_name = c('CMP_RNA_1','CMP_RNA_2','CFUE_RNA_1','CFUE_RNA_2'),
  group = c('CMP', 'CMP', 'CFUE', 'CFUE')
  )
colData$group = as.factor(colData$group)
designFormula <- "~ group"
```

## DESeq2

```{r DESEQ2}
library(DESeq2)

#create a DESeq dataset object from the count matrix and the colData 
dds <- DESeqDataSetFromMatrix(countData = countData, 
                              colData = colData, 
                              design = as.formula(designFormula))
#print dds object to see the contents
print(dds)

#For each gene, we count the total number of reads for that gene in all samples 
#and remove those that don't have at least 1 read. 
dds <- dds[ rowSums(DESeq2::counts(dds)) > 1, ]

dds <- DESeq(dds)

#compute the contrast for the 'group' variable where 'CTRL' 
#samples are used as the control group. 
DEresults = results(dds, contrast = c("group", 'CMP', 'CFUE'))
#sort results by increasing p-value
DEresults_print <- DEresults[order(DEresults$pvalue)[1:10],]
print(DEresults)
```


## Diagnostic plots
MA plot: Note that most points fall on the horizontal zero line, which means most genes are not differentially expressed.

P-value plot: It is also important to observe the distribution of raw p-values. We expect to see a peak around low p-values and a uniform distribution at P-values above 0.1. Otherwise, adjustment for multiple testing does not work and the results are not meaningful.
```{r diag plots}
#MA plot
DESeq2::plotMA(object = dds, ylim = c(-5, 5))

#Pvalue plot
library(ggplot2)
ggplot(data = as.data.frame(DEresults), aes(x = pvalue)) + 
  geom_histogram(bins = 100)

#PCA plot
rld <- rlog(dds)
DESeq2::plotPCA(rld, ntop = 500, intgroup = 'group') + 
  ylim(-50, 50) + theme_bw()

#RLE Plot
library(EDASeq)
par(mfrow = c(1, 1))
plotRLE(DESeq2::counts(dds, normalized = TRUE), 
        outline=FALSE, ylim=c(-4, 4), 
        col = as.numeric(colData$group), 
        main = 'Normalized Counts')
```

## Limma-VOOM
```{r limma voom}
library(edgeR)
#Create DGEList object
d0 <- DGEList(countData)

#Add normalizing factors
d0 <- calcNormFactors(d0)

#Drop low-expressed genes
cutoff <- 5
drop <- which(apply(cpm(d0), 1, max) < cutoff)
d <- d0[-drop,] 
dim(d) # number of genes left


group = c('CMP', 'CMP', 'CFUE', 'CFUE')

mm <- model.matrix(~group)
colnames(mm) <- gsub("group", "", colnames(mm))
print(mm)

y <- voom(d, mm, plot = T)
fit <- lmFit(y, mm)
head(coef(fit))

tmp <- eBayes(fit)

top.table <- topTable(tmp, sort.by = "P", n = Inf)
lv_dif_ex_genes = rownames(head(top.table, 1000))
```

```{r lv deseq2 compare}
library(VennDiagram)
library(gridExtra)
# extract differential expression results
DEresults <- results(dds, contrast = c('group', 'CMP', 'CFUE'))

#remove genes with NA values 
DE <- DEresults[!is.na(DEresults$padj),]
lowest_pvalue_indexes <- order(DE@listData$pvalue)[1:1000]
DE2_dif_ex_genes = DE@rownames[lowest_pvalue_indexes]

# Create a Venn diagram
venn.plot <- venn.diagram(
  x = list(lv_dif_ex_genes, DE2_dif_ex_genes),
  category.names = c("Limma-voom" , "DESEQ2"),
  filename = '#venn_diagramm.png',
  output=TRUE
)

```

## GO term analysis

Simply run this code on a set of genes that are significantly upregulated and then same with downregulated

#### Find the upregulated and downregulated  gene descriptions
```{r GO upreg}
library(DESeq2)
library(gprofiler2)
library(knitr)
# extract differential expression results
DEresults <- results(dds, contrast = c('group', 'CMP', 'CFUE'))

#remove genes with NA values 
DE <- DEresults[!is.na(DEresults$padj),]
#select genes with adjusted p-values below 0.1
DE <- DE[DE$padj < 0.1,]
#select genes with log2 fold change above 1 (two-fold change)
up_reg_DE <- DE[DE$log2FoldChange > 1,]
dn_reg_DE <- DE[DE$log2FoldChange < -1,]

#get the list of upregulated genes of interest
up_reg_gene_names <- rownames(up_reg_DE)
up_reg_gene_names <- sapply(strsplit(up_reg_gene_names, "\\."), function(x) x[[1]])
up_reg_gene_names <- unique(up_reg_gene_names)

#get the list of downregulated genes of interest
dn_reg_gene_names <- rownames(dn_reg_DE)
dn_reg_gene_names <- sapply(strsplit(dn_reg_gene_names, "\\."), function(x) x[[1]])
dn_reg_gene_names <- unique(dn_reg_gene_names)

#calculate enriched GO terms
up_go_response <- gost(query = up_reg_gene_names, 
                     organism = 'mmusculus',
                  sources = c("GO"))
dn_go_response <- gost(query = dn_reg_gene_names, 
                     organism = 'mmusculus',
                  sources = c("GO"))

# gostplot(up_go_response, capped=FALSE)
up_go_results = up_go_response$result
dn_go_results = dn_go_response$result
up_go_results <- up_go_results[order(up_go_results$p_value),]
dn_go_results <- dn_go_results[order(dn_go_results$p_value),]
# up_go_results <- up_go_results[up_go_results$intersection_size < 100,]

kable(up_go_results[1:10,c(3:11)])

kable(dn_go_results[1:10,c(3:11)])

# gostplot(up_go_response, capped = FALSE)
# gostplot(dn_go_response, capped = FALSE)
```
